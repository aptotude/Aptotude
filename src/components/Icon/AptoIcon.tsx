import React from 'react';
import classNames from 'classnames';
import './aptoIcon.scss';
import capitalize from '../../utils/capitalize';
import { StandardTypes } from '../../utils/standardTypes';
const COMPONENT_PREFIX = 'AptoIcon';

export type AptoIconColors =
  | 'white'
  | 'blue'
  | 'orange'
  | 'gray'
  | 'lightGray'
  | undefined;
export type AptoIconStatus = 'warning' | 'danger' | undefined;
export type AptoIconIcon =
  | React.ComponentClass<any>
  | React.StatelessComponent<any>;
export type AptoIconSize =
  | 1
  | 2
  | 3
  | 4
  | 5
  | 6
  | '1'
  | '2'
  | '3'
  | '4'
  | '5'
  | '6'
  | undefined;
export type AptoIconCircle = boolean | undefined;
export type AptoIconInline = boolean | undefined;

interface AptoIconProps extends StandardTypes {
  circleColor?: AptoIconColors;
  color?: AptoIconColors;
  status?: AptoIconStatus;
  inline?: AptoIconInline;
  circle?: AptoIconCircle;
  size?: AptoIconSize;
  icon: AptoIconIcon;
  onClick?: (event: React.MouseEvent<HTMLSpanElement>) => void;
}

class AptoIcon extends React.Component<AptoIconProps> {
  public static defaultProps = {
    inline: false,
    circle: false,
    size: 3
  };

  public render() {
    const {
      inline,
      circle,
      size,
      color,
      circleColor,
      icon: Component,
      status,
      className
    } = this.props;
    const classes = classNames(
      COMPONENT_PREFIX,
      `${COMPONENT_PREFIX}--size${size}`,
      inline ? `${COMPONENT_PREFIX}--inline` : null,
      circle ? `${COMPONENT_PREFIX}--circle` : null,
      status ? `${COMPONENT_PREFIX}--status${capitalize(status)}` : null,
      color ? `${COMPONENT_PREFIX}--color${capitalize(color)}` : null,
      circleColor
        ? `${COMPONENT_PREFIX}--circleColor${capitalize(circleColor)}`
        : null,
      className
    );

    const svgProps = {
      height: '100%',
      width: '100%',
      preserveAspectRatio: 'xMidYMid meet',
      focusable: 'false'
    };
    return (
      <span className={classes}>
        <Component {...svgProps} />
        {status && <span className="AptoIcon-status">!</span>}
      </span>
    );
  }
}

export { AptoIcon };
