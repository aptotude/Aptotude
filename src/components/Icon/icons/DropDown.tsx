import React from 'react';

const SvgDropDown = (props: React.SVGProps<SVGSVGElement>) => (
  <svg viewBox="0 0 24 24" {...props}>
    <path d="M7 10l5 5 5-5z" />
    <path d="M0 0h24v24H0z" fill="none" />
  </svg>
);

export default SvgDropDown;
