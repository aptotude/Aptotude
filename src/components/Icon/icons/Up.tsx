import React from 'react';

const SvgUp = (props: React.SVGProps<SVGSVGElement>) => (
  <svg viewBox="0 0 24 24" {...props}>
    <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
    <path d="M0 0h24v24H0z" fill="none" />
  </svg>
);

export default SvgUp;
