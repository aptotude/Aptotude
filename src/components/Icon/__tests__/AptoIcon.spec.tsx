import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { person } from '@apto/icons';
import { AptoIcon } from '../AptoIcon';

afterEach(() => {
  cleanup();
});

describe('AptoIcon', () => {
  it('Renders', () => {
    const { container } = render(<AptoIcon icon={person} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('color', () => {
    it('renders white', () => {
      const { container } = render(<AptoIcon color="white" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('renders blue', () => {
      const { container } = render(<AptoIcon color="blue" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('renders orange', () => {
      const { container } = render(<AptoIcon color="orange" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('renders gray', () => {
      const { container } = render(<AptoIcon color="gray" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('renders lightGray', () => {
      const { container } = render(
        <AptoIcon color="lightGray" icon={person} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('circle color', () => {
    it('renders white', () => {
      const { container } = render(
        <AptoIcon circleColor="white" icon={person} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    it('renders blue', () => {
      const { container } = render(
        <AptoIcon circleColor="blue" icon={person} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    it('renders orange', () => {
      const { container } = render(
        <AptoIcon circleColor="orange" icon={person} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    it('renders gray', () => {
      const { container } = render(
        <AptoIcon circleColor="gray" icon={person} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    it('renders lightGray', () => {
      const { container } = render(
        <AptoIcon circleColor="lightGray" icon={person} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('status', () => {
    it('warning', () => {
      const { container } = render(<AptoIcon status="warning" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('danger', () => {
      const { container } = render(<AptoIcon status="danger" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('size', () => {
    it('1', () => {
      const { container } = render(<AptoIcon size="1" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('2', () => {
      const { container } = render(<AptoIcon size="2" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('3', () => {
      const { container } = render(<AptoIcon size="3" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('4', () => {
      const { container } = render(<AptoIcon size="4" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('5', () => {
      const { container } = render(<AptoIcon size="5" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    it('6', () => {
      const { container } = render(<AptoIcon size="6" icon={person} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('renders circle', () => {
    const { container } = render(<AptoIcon circle icon={person} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders inline', () => {
    const { container } = render(<AptoIcon inline icon={person} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders extra class', () => {
    const { container } = render(<AptoIcon className="foooo" icon={person} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
