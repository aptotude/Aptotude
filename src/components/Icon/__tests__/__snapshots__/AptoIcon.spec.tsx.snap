// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoIcon Renders 1`] = `
<span
  class="AptoIcon AptoIcon--size3"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon circle color renders blue 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--circleColorBlue"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon circle color renders gray 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--circleColorGray"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon circle color renders lightGray 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--circleColorLightGray"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon circle color renders orange 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--circleColorOrange"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon circle color renders white 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--circleColorWhite"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon color renders blue 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--colorBlue"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon color renders gray 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--colorGray"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon color renders lightGray 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--colorLightGray"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon color renders orange 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--colorOrange"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon color renders white 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--colorWhite"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon renders circle 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--circle"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon renders extra class 1`] = `
<span
  class="AptoIcon AptoIcon--size3 foooo"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon renders inline 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--inline"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon size 1 1`] = `
<span
  class="AptoIcon AptoIcon--size1"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon size 2 1`] = `
<span
  class="AptoIcon AptoIcon--size2"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon size 3 1`] = `
<span
  class="AptoIcon AptoIcon--size3"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon size 4 1`] = `
<span
  class="AptoIcon AptoIcon--size4"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon size 5 1`] = `
<span
  class="AptoIcon AptoIcon--size5"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon size 6 1`] = `
<span
  class="AptoIcon AptoIcon--size6"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;

exports[`AptoIcon status danger 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--statusDanger"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
  <span
    class="AptoIcon-status"
  >
    !
  </span>
</span>
`;

exports[`AptoIcon status warning 1`] = `
<span
  class="AptoIcon AptoIcon--size3 AptoIcon--statusWarning"
>
  <svg
    focusable="false"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
  <span
    class="AptoIcon-status"
  >
    !
  </span>
</span>
`;
